### PR TITLE
Adjusted Middleware Provider Objects into Navymazing Style.

### PR DESCRIPTION
**Pull request by hhovsepy**
Did changes for all objects, all navigation and validation methods.
Removed several odd methods used in old navigation stuff.

{{pytest: cfme/tests/middleware --use-provider hawkular }}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 0d19662438479753b52f06ee09630bc86235a13a:

Lint OK! :cake: :punch: :cookie: .


**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

File: cfme/middleware/__init__.py [hunk 1]
```diff
@@ -6,9 +6,8 @@
 from cfme.common import Validatable, SummaryMixin
 from cfme.fixtures import pytest_selenium as sel
 from cfme.web_ui import toolbar as tb, Form, fill, flash, FileInput, Input
-from cfme.web_ui import CFMECheckbox, Select
+from cfme.web_ui import CFMECheckbox, Select, InfoBlock
```
Comment (mshriver) [inline cfme/middleware/__init__.py]: Can you please collapse all of these into a single list, formatted like: 

```python
from cfme.web_ui import (
    fill, flash, InfoBlock, [etc] )

File: cfme/middleware/datasource.py [hunk 2]
```diff
@@ -6,6 +8,8 @@
 from cfme.web_ui.menu import nav, toolbar as tb
 from mgmtsystem.hawkular import CanonicalPath
 from utils import attributize_string
+from utils.appliance import Navigatable, CurrentAppliance
```
Comment (mshriver) [inline cfme/middleware/datasource.py]: I believe you can remove CurrentAppliance import and setting the appliance attribute for all of these classes.

File: cfme/middleware/datasource.py [hunk 3]
```diff
@@ -68,37 +66,34 @@ def _get_datasources_page(provider=None, server=None):
...
     MiddlewareDatasource class provides details on datasource page.
     Class methods available to get existing datasources list
-
     Args:
         name: Name of the datasource
         provider: Provider object (HawkularProvider)
         nativeid: Native id (internal id) of datasource
         server: Server object of the datasource (MiddlewareServer)
         properties: Datasource driver name, connection URL and JNDI name
         db_id: database row id of datasource
```
Comment (mshriver) [inline cfme/middleware/datasource.py]: Should add appliance to your doc string here.

**Comment by dajoRH** [comment]
Lint report for commit f0eba7397e15be7eb9b13a9fe9e4eccd7fe5651f:

Lint OK! :cake: :punch: :cookie: .


**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Comment by hhovsepy** [comment]
failures caused by simultaneous test run by another job

**Comment by hhovsepy** [comment]
@mshriver have changed according to review comments

**Comment by mshriver** [comment]
A general comment, and I know this is a bit pedantic, but we would like to keep all of our imports ordered by built-in, imported, and local, with each section arranged alphabetically and a space between. I realize in some of these files you've only added some imports, but re-arranging them in this PR would be great! I don't believe we consider it a blocking issue though.

Example:
```python
import re

from navmazing import NavigateToSibling

from utils.appliance.implementiations.ui import navigate_to

**Pull request merged**